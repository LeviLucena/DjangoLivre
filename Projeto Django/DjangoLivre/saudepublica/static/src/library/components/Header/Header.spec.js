import { shallowMount } from '@vue/test-utils'
import BrHeader from './Header.ce.vue'

describe('Header', () => {
  test('it renders Header component', () => {
    const wrapper = shallowMount(BrHeader)
    expect(wrapper.classes('br-header')).toBe(true)
  })

  test('it renders logo image and signature', () => {
    const link = 'https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-positive.png'
    const assinatura = 'Assinatura da Empresa'

    const wrapper = shallowMount(BrHeader, {
      props: {
        image: link,
        signature: assinatura,
      },
    })

    expect(wrapper.find("img[src='" + link + "']").exists()).toBe(true)
    expect(wrapper.find('.header-sign').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it renders logo image with small density', () => {
    const link = 'https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-positive.png'
    const wrapper = shallowMount(BrHeader, {
      props: {
        image: link,
        density: 'small',
      },
    })

    expect(wrapper.find('.br-header.small').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it renders Menu', async () => {
    const wrapper = shallowMount(BrHeader, {
      propsData: {
        hasMenu: true,
      },
    })
    expect(wrapper.find('.header-menu-trigger').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it renders Title and subtitle', async () => {
    const wrapper = shallowMount(BrHeader, {
      props: {
        title: 'Título do Header',
        subtitle: 'Subtítulo do Header',
      },
    })
    expect(wrapper.find('.header-title').exists()).toBe(true)
    expect(wrapper.find('.header-subtitle').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('set prop container-fluid', () => {
    const wrapper = shallowMount(BrHeader, {
      props: {
        containerFluid: true,
      },
    })
    expect(wrapper.find('.container-fluid').exists()).toBe(true)
  })

  test('set sticky header', () => {
    const spyOnStickyChange = jest.spyOn(BrHeader.methods, 'onStickyChange')
    const wrapper = shallowMount(BrHeader, {
      props: {
        isSticky: true,
        hasSearch: true,
      },
    })
    expect(wrapper.find('.br-header.sticky.compact').exists()).toBe(false)
    window.scrollY = 1000
    window.dispatchEvent(new Event('scroll'))
    wrapper.vm.$nextTick(() => {
      expect(spyOnStickyChange).toHaveBeenCalled()
      expect(wrapper.find('.br-header.sticky.compact').exists()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  test('check if divider desapears on mobile', () => {
    const link = 'https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-positive.png'
    const assinatura = 'Assinatura da Empresa'

    const wrapper = shallowMount(BrHeader, {
      props: {
        image: link,
        signature: assinatura,
      },
    })
    expect(wrapper.find('br-divider-stub').exists()).toBe(true)

    global.innerWidth = 200
    global.dispatchEvent(new Event('resize'))

    expect(wrapper.find('.br-divider').exists()).toBe(false)
  })

  test('set small image-size', () => {
    const link = 'https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-positive.png'
    const wrapper = shallowMount(BrHeader, {
      props: {
        imageSize: 'small',
        image: link,
      },
    })
    expect(wrapper.find('.small-img').exists()).toBe(true)
  })
})
