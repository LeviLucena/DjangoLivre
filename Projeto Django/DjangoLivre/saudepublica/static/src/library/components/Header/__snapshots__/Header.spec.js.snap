// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header it renders Menu 1`] = `
<header
  class="br-header mb-4"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <div
          class="header-menu-trigger"
        >
          
          <br-button-stub
            active="false"
            aria-label="Menu"
            block="false"
            circle="true"
            data-target="#main-navigation"
            data-toggle="menu"
            density="small"
            disabled="false"
            id="navigation"
            inverted="false"
            loading="false"
          />
          
        </div>
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <!--v-if-->
    </div>
  </div>
</header>
`;

exports[`Header it renders Title and subtitle 1`] = `
<header
  class="br-header mb-4"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <div
            class="header-title"
          >
            
            Título do Header
            
          </div>
          <div
            class="header-subtitle"
          >
            Subtítulo do Header
          </div>
        </div>
      </div>
      <!--v-if-->
    </div>
  </div>
</header>
`;

exports[`Header it renders logo image and signature 1`] = `
<header
  class="br-header mb-4"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <img
          alt="logo"
          class="large-img"
          src="https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-positive.png"
        />
        <br-divider-stub
          class="mx-half mx-sm-1"
          dashed="false"
          vertical="true"
        />
        <div
          class="header-sign"
        >
          Assinatura da Empresa
        </div>
      </div>
      
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <!--v-if-->
    </div>
  </div>
</header>
`;

exports[`Header it renders logo image with small density 1`] = `
<header
  class="br-header mb-4 small"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <img
          alt="logo"
          class="large-img"
          src="https://cdngovbr-ds.estaleiro.serpro.gov.br/design-system/images/logo-positive.png"
        />
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <!--v-if-->
    </div>
  </div>
</header>
`;

exports[`Header set sticky header 1`] = `
<header
  class="br-header mb-4 sticky compact"
  data-sticky="true"
>
  <div
    class="container-lg"
  >
    <div
      class="header-top"
    >
      <div
        class="header-logo"
      >
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      
      
    </div>
    <div
      class="header-bottom"
    >
      <div
        class="header-menu"
      >
        <!--v-if-->
        <div
          class="header-info"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
      <div
        class="header-search"
      >
        <br-header-search-stub
          id="searchbox15"
          label="Texto da pesquisa"
          placeholder="O que você procura?"
        />
      </div>
    </div>
  </div>
</header>
`;
